exec/java-exec/src/main/java/org/apache/drill/exec/store/ischema/Records.java

```diff
@@ -750,10 +751,8 @@ public Partition(String catalog, String schemaName, String value, PartitionMetad
       this.LAST_MODIFIED_TIME = convertToTimestamp(partitionMetadata.getLastModifiedTime());
     }
 
-    public static List<Partition> fromSegment(String catalog, String schemaName, SegmentMetadata segmentMetadata) {
-      return segmentMetadata.getPartitionValues().stream()
-        .map(value -> new Partition(catalog, schemaName, value, segmentMetadata))
-        .collect(Collectors.toList());
+    public static Partition fromSegment(String catalog, String schemaName, SegmentMetadata segmentMetadata) {
```

Comment:
Please revisit this change, I am not sure it's correct.

Reply:
You are right, changed the approach to populate segment partition values correctly. Thanks a lot for pointing this.